Engine/src/Engine/Scene.cpp

```diff
@@ -335,66 +384,72 @@ namespace Engine
 #pragma region Renderable Entities
 	
 	//Return the VBO for sprites. If it doesn't exist, create it.
-	GLuint Scene::getSpriteVBO() 
+	GLuint Scene::getVBO(RenderableType type) 
 	{
-		if(!m_createdVBO);
-		{
-			m_createdVBO = true;
+		GLuint vbo;
+		glGenBuffers(1, &vbo);
+		glBindBuffer(GL_ARRAY_BUFFER, vbo);
 
+		if (type == SPRITE) 
+		{
 			float vertices[] = 
 			{
 				// positions  // texture coords (UV coords)
-
 				0.f, 0.f, 0.f,  0.f, 0.f,  // top left
 				1.f, 0.f, 0.f,  1.f, 0.f,  // top right
 				1.f, 1.f, 0.f,  1.f, 1.f,  // bottom right
 				0.f, 1.f, 0.f,  0.f, 1.f,  // bottom left
 			};
-			
-			
-			glGenBuffers(1, &m_spriteVBO);
-			glBindBuffer(GL_ARRAY_BUFFER, m_spriteVBO);
-
 			glBufferData(GL_ARRAY_BUFFER, sizeof(vertices), vertices, GL_DYNAMIC_DRAW);
 		}
-
-		return m_spriteVBO;
+		else if (type == RECTANGLE) 
+		{
+			float vertices[] = 
+			{
+				// positions 
+				0.f, 0.f, 0.f,  // top left
+				1.f, 0.f, 0.f,  // top right
+				1.f, 1.f, 0.f,  // bottom right
+				0.f, 1.f, 0.f   // bottom left
+			};
+			glBufferData(GL_ARRAY_BUFFER, sizeof(vertices), vertices, GL_DYNAMIC_DRAW);
+		}
+		else 
+		{
+			GE_CORE_ERROR("A vbo was declared with incorrect vertex data.");
+		}
+		return vbo;
 	}
 
 	//Return the VAO for sprites. If it doesn't exist, create it.
-	GLuint Scene::getSpriteVAO()
+	GLuint Scene::getVAO()
 	{
-		if (!m_createdVAO)
-		{
-			m_createdVAO = true;
-
-			glGenVertexArrays(1, &m_spriteVAO);
-			glBindVertexArray(m_spriteVAO);
-		}
-		
-		return m_spriteVAO;
+		GLuint vao;
+		glGenVertexArrays(1, &vao);
+		glBindVertexArray(vao);
+		return vao;
 	}
 
 	//Return the IBO for sprites. If it doesn't exist, create it.
-	GLuint Scene::getSpriteIBO()
+	GLuint Scene::getIBO(RenderableType type)
 	{
-		if (!m_createdIBO);
-		{
-			m_createdIBO = true;
-			
+		GLuint ibo;
+		glGenBuffers(1, &ibo);
+		glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, ibo);
+
+		if (type == RenderableType::RECTANGLE) {
```

Comment:
our current rule is all caps for const global variables.
We don't have any style guidelines for enums, but in serializer, I kind of set one by grabbing two letters from enum and then the name of the enum since I've seen that convention elsewhere. In this case, it would be RT_Rectangle.
If that sounds like a good standard we can add it.